src/main/java/org/mitre/synthea/export/HL7V2Exporter.java

```diff
@@ -0,0 +1,626 @@
+
+public class HL7V2Exporter {
+
+    private static final String HL7_VERSION = "2.5.1";
+    private static final String CW_NAMESPACE_ID = "CWNS";
+    private static final String MSG_TYPE = "ADT";
+    private static final String MSG_EVENT_TYPE = "A01";
+
+    private ADT_A01 adt;
+    private TreeMap<String, String> customSegs = new TreeMap();
+
+    public static HL7V2Exporter getInstance() {
+        return new HL7V2Exporter();
+    }
+
+    public String export(Person person, long time) {
+        // create a super encounter... this makes it easier to access
+        // all the Allergies (for example) in the export templates,
+        // instead of having to iterate through all the encounters.
+        Encounter superEncounter = person.record.new Encounter(time, "super");
+        for (Encounter encounter : person.record.encounters) {
+            if (encounter.start <= time) {
+                superEncounter.observations.addAll(encounter.observations);
+                superEncounter.reports.addAll(encounter.reports);
+                superEncounter.conditions.addAll(encounter.conditions);
+                superEncounter.allergies.addAll(encounter.allergies);
+                superEncounter.procedures.addAll(encounter.procedures);
+                superEncounter.immunizations.addAll(encounter.immunizations);
+                superEncounter.medications.addAll(encounter.medications);
+                superEncounter.careplans.addAll(encounter.careplans);
+                superEncounter.imagingStudies.addAll(encounter.imagingStudies);
+            } else {
+                break;
+            }
+        }
+
+        // The export templates fill in the record by accessing the attributes
+        // of the Person, so we add a few attributes just for the purposes of export.
+        person.attributes.put("UUID", UUID.randomUUID().toString());
+        person.attributes.put("ehr_encounters", person.record.encounters);
+        person.attributes.put("ehr_observations", superEncounter.observations);
+        person.attributes.put("ehr_reports", superEncounter.reports);
+        person.attributes.put("ehr_conditions", superEncounter.conditions);
+        person.attributes.put("ehr_allergies", superEncounter.allergies);
+        person.attributes.put("ehr_procedures", superEncounter.procedures);
+        person.attributes.put("ehr_immunizations", superEncounter.immunizations);
+        person.attributes.put("ehr_medications", superEncounter.medications);
+        person.attributes.put("ehr_careplans", superEncounter.careplans);
+        person.attributes.put("ehr_imaging_studies", superEncounter.imagingStudies);
+        person.attributes.put("time", time);
+        person.attributes.put("race_lookup", RaceAndEthnicity.LOOK_UP_CDC_RACE);
+        person.attributes.put("ethnicity_lookup", RaceAndEthnicity.LOOK_UP_CDC_ETHNICITY_CODE);
+        person.attributes.put("ethnicity_display_lookup", RaceAndEthnicity.LOOK_UP_CDC_ETHNICITY_DISPLAY);
```

Comment:
I'm fairly certain that this big block of code is all related to the Freemarker exporters. They use these attributes in the templates, but you don't need that here.

Reply:
Perfect! All that code can go. I should obviously be using **superEncounter** and **person.record.encounters**, etc. I mean, it's all right there in memory (the longitudinal model that is).